src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java

```diff
@@ -629,36 +650,31 @@ protected static String parseJavadocMemory( String memory )
             throw new IllegalArgumentException( "The memory could not be null." );
         }
 
-        Pattern p = Pattern.compile( "^\\s*(\\d+)\\s*?\\s*$" );
-        Matcher m = p.matcher( memory );
+        Matcher m = PARSE_JAVADOC_MEMORY_PATTERN_0.matcher( memory );
         if ( m.matches() )
         {
             return m.group( 1 ) + "m";
         }
 
-        p = Pattern.compile( "^\\s*(\\d+)\\s*k(b)?\\s*$", Pattern.CASE_INSENSITIVE );
-        m = p.matcher( memory );
+        m = PARSE_JAVADOC_MEMORY_PATTERN_1.matcher( memory );

```

Comment:
in general, we should avoid reusing local variables for different objects

Reply:
@elharo done.
btw, this reusing is by original codes, and I don't like the reuse either.
now fixed.